### PR TITLE
Adding to and updating Documentdb Templates

### PR DESCRIPTION
**Pull request by dmakwana**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added new quickstart template for multi-region database accounts
* Updated existing templates with new 'locations' array required in payload

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dmakwana__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (dimakwan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by dmakwana** [comment]
This doesn't seem to failing due to my changes.

File: 101-documentdb-account-create-multi-region-account/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "databaseAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "The DocumentDB database account name."
+      }
+    },
+    "writeRegionLocation": {
+      "type": "string",
```
Comment (gatneil) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: Please add a default value

File: 101-documentdb-account-create-multi-region-account/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      "metadata": {
+        "description": "The location of the write region of the database account."
+      }
+    },
+    "readRegionLocation": {
+      "type": "string",
```
Comment (gatneil) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: same

File: 101-documentdb-account-create-multi-region-account/azuredeploy.json (hunk 1 above)
Comment (dmakwana) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: Doesn't the azuredeploy.parameters.json file contain the default values?

File: 101-documentdb-account-create-multi-region-account/azuredeploy.json (hunk 2 above)
Comment (dmakwana) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: Doesn't the azuredeploy.parameters.json file contain the default values?

File: 101-documentdb-account-create-multi-region-account/azuredeploy.json (hunk 1 above)
Comment (gatneil) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: I've been getting questions about this a lot lately; threw together a wiki page about it that hopefully clears things up: https://github.com/Azure/azure-quickstart-templates/wiki/defaultValue-vs.-azuredeploy.parameters.json-vs.-GEN-*

Comment (gatneil) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: Thinking about this a bit more, I think this is debatable either way; personally I'd prefer to have the default values for the regions in the "defaultValue" of the azuredeploy.json file for the reasons described in that wiki page, but I'll leave the decision up to you which you prefer :)

Comment (dmakwana) [inline 101-documentdb-account-create-multi-region-account/azuredeploy.json]: Thanks @gatneil!

Could we keep it as is then?

**Comment by dmakwana** [comment]
ping

**Comment by dmakwana** [comment]
@gatneil ping

**Comment by gatneil** [comment]
looks good; merging

**Pull request merged**